src/pow.rs

```diff
@@ -182,9 +183,13 @@ mod float_impls {
 /// assert_eq!(pow(6u8, 3), 216);
 /// ```
 #[inline]
-pub fn pow<T: Clone + One + Mul<T, Output = T>>(mut base: T, mut exp: usize) -> T {
+pub fn pow<T: Clone + Zero + One + Mul<T, Output = T>>(mut base: T, mut exp: usize) -> T {
     if exp == 0 {
-        return T::one();
+        return if base.is_zero() {
+            panic!("0⁰ is undefined")
```

Comment:
I would restrain myself from using Unicode in panic messages as it can be problematic on platforms that do not support Unicode in CLI (aka Windows).

Reply:
do we want: 
* 0^0
* pow(0, 0)
* any other?

Reply:
I would go with `pow(0, 0)` as this is what user wrote. `0^0` is common syntax but in this case can be confused with XOR operator.